common/common/src/main/java/io/helidon/common/Version.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.common;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Simple class to provide version information.
+ */
+public class Version {
```

Comment:
I'd rather avoid the static design.
How about this instead:
- make Version a lazy initialized singleton
- make Version.getInstance private
- create static methods to access the properties, e.g. 
```java
public static String getVersion(){
    return getInstance().version;
}
```

Reply:
This class does not need to be instantiated at all.
Static initialization is IMHO OK.
I would just change the method `getVersion()` to be static.

Reply:
static initializer are evil. Using config alone would mean having this class load the properties even if not used.

Reply:
The class is not loaded until used. So if you use Config, this will never get loaded.
It gets loaded only when you start a webserver or when you use it in your code.
So normal statically initialized singleton is absolutely fine, as well as static initialization of the properties.

Reply:
Yup, I forgot about that ; was desperately trying to argue against static initializer ;)

Reply:
Implementation has now changed to generated code and true constants. No more static initializer.